### PR TITLE
OAK-10259: oak-core tests running out of memory

### PR DESCRIPTION
**Pull request by mreutegg**
Clear inline mocks at the end of a test

### CODE REVIEW

File: oak-core/src/test/java/org/apache/jackrabbit/oak/security/authentication/user/LoginModuleImplTest.java [hunk 1]
```diff
@@ -106,7 +105,6 @@ public class LoginModuleImplTest extends AbstractSecurityTest {
     @Override
     public void after() throws Exception {
         try {
-            clearInvocations(monitor);
```
Comment (anchela) [inline oak-core/src/test/java/org/apache/jackrabbit/oak/security/authentication/user/LoginModuleImplTest.java]: i checked in my local oak and there are more usages of 'Mockito.clearInvocations()' in subclasses of the AbstractSecurityTest both in oak-core and in other security modules.

i am curious to understand why you remove this one and not the others.

**Review by anchela — CHANGES_REQUESTED** [review]
hi @mreutegg , the patch looks good. but i have 1 question regarding the removal of clearInvocations in the LoginModuleImplTest. 
if that is needed we might have to remove the other usages as well?

File: oak-core/src/test/java/org/apache/jackrabbit/oak/security/authentication/user/LoginModuleImplTest.java (hunk 1 above)
Comment (mreutegg) [reply oak-core/src/test/java/org/apache/jackrabbit/oak/security/authentication/user/LoginModuleImplTest.java]: With this line the test failed with:
```
org.mockito.exceptions.misusing.NotAMockException: Argument should be a mock, but is: class org.apache.jackrabbit.oak.spi.security.authentication.LoginModuleMonitor$MockitoMock$AIxpL0Xb
	at org.apache.jackrabbit.oak.security.authentication.user.LoginModuleImplTest.after(LoginModuleImplTest.java:110)
```
It seems the sequence how multiple After methods on the same class are called is determined by method name (lexicographic ordering?). It means the existing `after()` method is called first and then the newly introduced `clearMocks()`.

I will revert the change in `LoginModuleImplTest` and move `Mockito.framework().clearInlineMocks()` to the existing `after()` method. The changes to `LoginContextProviderImplTest` will still be necessary though.

**Comment by mreutegg** [comment]
@anchela WDYT now?

**Review by anchela — APPROVED** [review]
lgtm (but i didn't build oak with the modifications nor did i verify that the oom is fixed)... but the patch looks reasonable. hope that helps.

**Pull request merged**